### PR TITLE
modify unhangRange logic

### PR DESCRIPTION
**Pull request by suilang**
**Description**
As shown in the test case，if there has two blocks, and each block only has one inline void element,  like <block><text/><inline void><text/></inline/><text/></block/>.  The anchor in first text of first line, the focus in the last text of last line.

When delete selected content, the function of unhangRange will change current range to the last text of the first line.

**Checks**
- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)
- [x] You've [added a changeset](https://github.com/atlassian/changesets/blob/master/docs/adding-a-changeset.md) if changing functionality. (Add one with `yarn changeset add`.)

### CODE REVIEW

**Comment by suilang** [comment]
There's a problem. I want to optimize this logic a little bit more, but several tests are going to fail anyway. But the current version is really error-free

**Review by dylans — APPROVED** [review]

File: packages/slate/src/interfaces/editor.ts [hunk 1]
```diff
@@ -1630,7 +1630,7 @@ export const Editor: EditorInterface = {
 
     for (const [node, path] of Editor.nodes(editor, {
       at: before,
-      match: Text.isText,
+      match: n => !Editor.isBlock(editor, n) && !Editor.isEditor(n),
```
Comment (dylans) [inline packages/slate/src/interfaces/editor.ts:1633]: Might be clearer if it was specific in what it is including, e.g.

Text.isText || Editor.isVoid || Editor.isInline ?

File: packages/slate/src/interfaces/editor.ts [hunk 2]
```diff
@@ -1639,8 +1639,8 @@ export const Editor: EditorInterface = {
         continue
       }
 
-      if (node.text !== '' || Path.isBefore(path, blockPath)) {
-        end = { path, offset: node.text.length }
+      if (!Text.isText(node) || Path.isBefore(path, blockPath)) {
```
Comment (nemanja-tosic) [inline packages/slate/src/interfaces/editor.ts:1642]: Question related to the comment on line 1633 - what are we targeting here? An inclusive list might be better. Alternatively, we should extract a variable that describes the conditional.

Comment (suilang) [reply packages/slate/src/interfaces/editor.ts:1642]: The conditions in the match function correspond to the following judgments. You're right, maybe we should revisit what does this function really want. Please do not deal with this PR for the time being. I will deal with the logic of this func again later if I have time

Comment (dylans) [reply packages/slate/src/interfaces/editor.ts:1642]: Meaning you would like us to land the PR and then improve this, or meaning just wait until you have more time?

I think it's a pretty important fix given some of the recent regressions, so I'll take a look soon if you don't have time.

Comment (suilang) [reply packages/slate/src/interfaces/editor.ts:1642]: I'm really sorry that I didn't have time to deal with these due to some unexpected things. I will close this PR first, and you can handle it first

**Pull request closed without merging**